gateway/server.go

```diff
@@ -406,6 +407,17 @@ func (gw *Gateway) setupGlobals() {
 		go gw.flushNetworkAnalytics(gw.ctx)
 	}
 
+	var oAuthTokensPurgeInterval time.Duration
+	if purgeInterval := gw.GetConfig().Private.OAuthTokensPurgeInterval; purgeInterval == 0 {
```

Comment:
FYI even if you embed Private as suggested, it allows using `.Private` like you do here, if you want to be more explicit that's fine.

I would suggest to move this scanner logic onto the type declaration, and provide a getter for the computation. This extends an already long function, so provide this as a `func(*Config)time.Duration` or similar to not extend this function further.